File: server/View/ViewModel.cs
using System;
using Newtonsoft.Json;

namespace Com.GitHub.GlowingPotato.IslandDecimation.Server.View {
    public class ViewModel : IViewModel {
        public bool HasUpdated {
            get {
                return UserIsland.HasUpdated || BattlingIslands.HasUpdated || Troops.HasUpdated || StateHasUpdated;
            }
        }

#region UserIsland
        [JsonProperty("userIsland")]
        public Island UserIsland;

        public bool ShouldSerializeUserIsland() {
            return UserIsland.HasUpdated;
        }
#endregion
#region BattlingIslands
        [JsonProperty("battlingIslands")]
        public ViewModelList<Island> BattlingIslands;

        public bool ShouldSerializeBattlingIslands() {
            return BattlingIslands.HasUpdatedAndClear();
        }
#endregion
#region Troops
        [JsonProperty("troops")]
        public ViewModelList<Troop> Troops;

        public bool ShouldSerializeTroops() {
            return Troops.HasUpdatedAndClear();
        }
#endregion
#region State
        ViewState state;
        bool StateHasUpdated = true;

        [JsonProperty("state")]
        public ViewState State {
            get {
                return state;
            }
            set {
                state = value;
                StateHasUpdated = true;
            }
        }

        public bool ShouldSerializeState() {
            bool val = StateHasUpdated;
            if (val) {
                StateHasUpdated = false;
            }
            return val;
        }
#endregion
    }
}

File: client/src/main/java/com/github/glowingpotato/islanddecimation/view/ViewModel.java
package com.github.glowingpotato.islanddecimation.view;

import java.util.ArrayList;

public class ViewModel {
	
	private ArrayList<Island> userIslands = new ArrayList<Island>();
	private ArrayList<Island> battlingIslands = new ArrayList<Island>();
	private ArrayList<Troop> troops = new ArrayList<Troop>();
	private ViewState state;
	
	public ViewModel(ArrayList<Island> userIslands, ArrayList<Island> battlingIslands, ArrayList<Troop> troops, ViewState state) {
		this.userIslands = userIslands;
		this.battlingIslands = battlingIslands;
		this.troops = troops;
		this.state = state;
	}
	
	public void update(ViewModel viewModel) {
		if(viewModel.getUserIslands() != null)
			setUserIslands(viewModel.getUserIslands());
		if(viewModel.getBattlingIslands() != null)
			setBattlingIslands(viewModel.getBattlingIslands());
		if(viewModel.getTroops() != null)
			setTroops(viewModel.getTroops());
		if(viewModel.getState() != null)
			setState(viewModel.getState());
	}

	public ArrayList<Island> getUserIslands() {
		return userIslands;
	}

	public void setUserIslands(ArrayList<Island> userIslands) {
		this.userIslands = userIslands;
	}

	public ArrayList<Island> getBattlingIslands() {
		return battlingIslands;
	}

	public void setBattlingIslands(ArrayList<Island> battlingIslands) {
		this.battlingIslands = battlingIslands;
	}

	public ArrayList<Troop> getTroops() {
		return troops;
	}

	public void setTroops(ArrayList<Troop> troops) {
		this.troops = troops;
	}

	public ViewState getState() {
		return state;
	}

	public void setState(ViewState state) {
		this.state = state;
	}
}
